src/core/tests/Test_AptitudePackageManager.py

```diff
@@ -349,6 +373,117 @@ def test_update_image_default_patch_mode_raises_exception(self):
+        backup_package_manager_ubuntu_pro_client_is_pro_working = package_manager.ubuntu_pro_client.is_pro_working
+        package_manager.ubuntu_pro_client.is_pro_working = self.mock_is_pro_working_return_true
+        backup_package_manager_is_minimum_required_python_installed = package_manager._AptitudePackageManager__is_minimum_required_python_installed
+        package_manager._AptitudePackageManager__is_minimum_required_python_installed = self.mock_minimum_required_python_installed_return_true
+
+        self.assertTrue(package_manager.check_pro_client_prerequisites())
+
+        package_manager.ubuntu_pro_client.is_pro_working = backup_package_manager_ubuntu_pro_client_is_pro_working
+        package_manager._AptitudePackageManager__is_minimum_required_python_installed = backup_package_manager_is_minimum_required_python_installed
+
+    def test_package_manager_instance_created_even_when_exception_thrown_in_pro(self):
+        package_manager = self.container.get('package_manager')
+        execution_config = self.container.get('execution_config')
+        backup_package_manager_ubuntu_pro_client_install_or_update_pro = UbuntuProClient.UbuntuProClient.install_or_update_pro
+        UbuntuProClient.UbuntuProClient.install_or_update_pro = self.mock_install_or_update_pro_raise_exception
+
+        obj = AptitudePackageManager.AptitudePackageManager(package_manager.env_layer, execution_config, package_manager.composite_logger, package_manager.telemetry_writer, package_manager.status_handler)
+
+        self.assertIsNotNone(obj)
+        self.assertIsNotNone(obj.ubuntu_pro_client)
+
+        UbuntuProClient.UbuntuProClient.install_or_update_pro = backup_package_manager_ubuntu_pro_client_install_or_update_pro
+
+    def test_is_reboot_pending_pro_client_success(self):
+        version_mock = MockVersionResult()
+        version_mock.mock_import_uaclient_version_module('version', 'mock_version')
+        reboot_mock = MockRebootRequiredResult()
+        reboot_mock.mock_import_uaclient_reboot_required_module('reboot_required', 'mock_reboot_required_return_no')
+        runtime = RuntimeCompositor(ArgumentComposer().get_composed_arguments(), True, Constants.APT)
+        backup_AptitudePackageManager__pro_client_prereq_met = runtime.package_manager._AptitudePackageManager__pro_client_prereq_met
+        runtime.package_manager._AptitudePackageManager__pro_client_prereq_met = True
+        self.assertFalse(runtime.package_manager.is_reboot_pending())
+
+        runtime.package_manager._AptitudePackageManager__pro_client_prereq_met = backup_AptitudePackageManager__pro_client_prereq_met
+        version_mock.mock_unimport_uaclient_version_module()
+        reboot_mock.mock_unimport_uaclient_reboot_required_module()
+
+    def test_is_reboot_pending_test_mismatch(self):
+        version_mock = MockVersionResult()
+        version_mock.mock_import_uaclient_version_module('version', 'mock_version')
```

Comment:
There is no need of version_mock because setting  runtime.package_manager._AptitudePackageManager__pro_client_prereq_met = True

Also, there is no need of reboot_mock as setting runtime.package_manager.ubuntu_pro_client.is_reboot_pending = self.mock_is_reboot_pending_returns_True

Reply:
Made the changes to remove the reboot_pending method mock. Done.,